core/src/test/scala/unit/kafka/log/LogTest.scala

```diff
@@ -1072,6 +1083,114 @@ class LogTest extends JUnitSuite {
 
     log.delete()
     assertEquals("The number of segments should be 0", 0, log.numberOfSegments)
-    assertEquals("The number of deleted segments shoud be zero.", 0, log.deleteOldSegments(_ => true))
+    assertEquals("The number of deleted segments should be zero.", 0, log.deleteOldSegments())
+  }
+
+
+  @Test
+  def shouldDeleteSizeBasedSegments(): Unit = {
+    val set = TestUtils.singleMessageSet("test".getBytes)
+    val (_, log: Log) = createLog(set.sizeInBytes, retentionBytes = set.sizeInBytes * 10)
+
+    // append some messages to create some segments
+    for (i <- 0 until 15)
+      log.append(set)
+
+    log.deleteOldSegments
+    assertEquals("should have 2 segments", 2,log.numberOfSegments)
+  }
+
+  @Test
+  def shouldNotDeleteSizeBasedSegmentsWhenUnderRetentionSize(): Unit = {
+    val set = TestUtils.singleMessageSet("test".getBytes)
+    val (_, log: Log) = createLog(set.sizeInBytes, retentionBytes = set.sizeInBytes * 15)
+
+    // append some messages to create some segments
+    for (i <- 0 until 15)
+      log.append(set)
+
+    log.deleteOldSegments
+    assertEquals("should have 3 segments", 3,log.numberOfSegments)
+  }
+
+  @Test
+  def shouldDeleteTimeBasedSegmentsReadyToBeDeleted(): Unit = {
+    val set = TestUtils.singleMessageSet("test".getBytes)
```

Comment:
Now that KIP-33 is committed, going forward, segments will be retained by timestamp in the message, not last modified time of the segment file. Could we insert messages with timestamp in this and subsequent tests?